api/workAllocation2/util.ts

```diff
@@ -155,64 +116,27 @@ export function prepareRoleApiRequest(locationId?: number): any {
 }
 
 /**
- * The below sets up actions on the tasks by directing them to respective methods
- * for each permission and concatenating any resulting lists
- * Note - Read, Refer and Owan are also permissions but currently have no respective method
- * Note - Concatenating suitable currently (might require changing if actions in few permissions)
- * @param permissions The list of permissions the user holds
+ * Aggregate permissions from the View Permissions Actions Matrix defined by business.
  * @param view This dictates which set of actions we should use.
- * @param isTaskAssigned Dummy parameter to test change based on whether task is assigned
+ * @param permissions The list of permissions the user holds.
+ * @return actionList:Action[] the list of total actions user holds.
  */
-export function getWATaskActions(permissions: TaskPermission[], view: ActionViews, isTaskAssigned: boolean): Action[] {
+export function getActionsByPermissions(view, permissions: []): Action[] {
   let actionList: Action[] = [];
   permissions.forEach(permission => {
     switch (permission) {
       case TaskPermission.MANAGE:
-        actionList = actionList.concat(getWATaskActionsForManage(view, isTaskAssigned));
```

Comment:
Why has isTaskAssigned been removed here? There needs to be a check that the task is assigned in order to get the correct actions. This looks like it will return the same for assigned and unassigned tasks for All Work. Because we're looping through the tasks, the check whether task is assigned needs to be internal to one of the methods in this util file.

Reply:
We don't need to check whether the task is assigned(isTaskAssigned), as the new mock APIs will call different apis to assigned or unassigned tasks. I will give you a demo on how it works.

Reply:
As discuss as we are not sure how the API returns assigned/unassigned tasks, we will keep this as simple as we can for the permission check.